tests/filecheck/dialects/riscv/riscv_ops_custom_syntax.mlir

```diff
@@ -0,0 +1,270 @@
+// RUN: xdsl-opt %s | filecheck %s
+"builtin.module"() ({
+  %0 = "riscv.get_register"() : () -> !riscv.reg<>
+  %1 = "riscv.get_register"() : () -> !riscv.reg<>
+  // CHECK: %{{.*}} = riscv.get_register : () -> (!riscv.reg<>)
+  // CHECK-NEXT: %{{.*}} = riscv.get_register : () -> (!riscv.reg<>)
+  // RV32I/RV64I: 2.4 Integer Computational Instructions
+
+  // Integer Register-Immediate Instructions
+  %addi = "riscv.addi"(%0) {"immediate" = 1 : i32}: (!riscv.reg<>) -> !riscv.reg<>
+  %slti = "riscv.slti"(%0) {"immediate" = 1 : i32}: (!riscv.reg<>) -> !riscv.reg<>
+  %sltiu = "riscv.sltiu"(%0) {"immediate" = 1 : i32}: (!riscv.reg<>) -> !riscv.reg<>
+  %andi = "riscv.andi"(%0) {"immediate" = 1 : i32}: (!riscv.reg<>) -> !riscv.reg<>
+  %ori = "riscv.ori"(%0) {"immediate" = 1 : i32}: (!riscv.reg<>) -> !riscv.reg<>
+  %xori = "riscv.xori"(%0) {"immediate" = 1 : i32}: (!riscv.reg<>) -> !riscv.reg<>
+  %slli = "riscv.slli"(%0) {"immediate" = 1 : i32}: (!riscv.reg<>) -> !riscv.reg<>
+  %srli = "riscv.srli"(%0) {"immediate" = 1 : i32}: (!riscv.reg<>) -> !riscv.reg<>
+  %srai = "riscv.srai"(%0) {"immediate" = 1 : i32}: (!riscv.reg<>) -> !riscv.reg<>
+  %lui = "riscv.lui"() {"immediate" = 1 : i32}: () -> !riscv.reg<>
+  %auipc = "riscv.auipc"() {"immediate" = 1 : i32}: () -> !riscv.reg<>
+  %mv = "riscv.mv"(%0) : (!riscv.reg<>) -> !riscv.reg<>
+  // CHECK-NEXT: %addi = riscv.addi %0, 1 : (!riscv.reg<>) -> (!riscv.reg<>)
```

Comment:
Can you try this, please? I think it would make it quite a bit more readable
```suggestion
  // CHECK-NEXT: %addi = riscv.addi %0, 1 : x? -> x?
```

Reply:
I don't think we strictly speaking need the types for the inputs, maybe there's some way to go without them, but maybe they're good for redundancy.

Reply:
I think `x?` is a good standin for an unallocated register, much better than None, which is weird in this language-agnostic context.

Reply:
I disagree that ```x?``` is good. You probably forget floating point registers and also other kind of registers such as vector registers in V-extension.

Reply:
Actually, only show the result type is good enough. Because we are using SSA here, for SSA arguments one can always go back to the lines that define the arguments for the type.

Reply:
I don't forget them, they are much less common, and we can find a good syntax for them also. For now I'd be happy with `riscv.freg<>` for floating point operations, although `f?` seems preferable also.

Reply:
Let's do this in a follow-up PR